src/libumf.def

```diff
@@ -14,6 +14,7 @@ EXPORTS
     umfFree
     umfGetIPCHandle
     umfGetLastFailedMemoryProvider
+    umfLevelZeroMemoryProviderOps
```

Comment:
This causes an error:
https://github.com/oneapi-src/unified-memory-framework/actions/runs/8705383479/job/23878531579?pr=435#step:6:71

Reply:
That's because there is no umfLevelZeroMemoryProviderOps symbol when umfLevelZeroMemoryProviderOps is not build (which is the case in that job). I would suggest to add `umfLevelZeroMemoryProviderOps` though cmake's configure_file only when L0 provider is build.

Basically you can add something like `@OPTIONAL_SYMBOLS@` to the libumf.def file and then set OPTIONAL_SYMBOLS variable in cmake to either "" or "umfLevelZeroMemoryProviderOps` and call configure_file("${CMAKE_CURRENT_SOURCE_DIR}/libumf.def"
		       "${CMAKE_CURRENT_BINARY_DIR}/libumf.def" @ONLY)

Reply:
Looks good! Could you just rename the original file to `libumf.def.in` - this way it will be clear that the file needs to be processed first.

Reply:
done